### PR TITLE
Add slack notifier to starter.yaml, skip notify for upgrade job, remo…

### PR DESCRIPTION
**Pull request by mszostok**
**Description**

Changes proposed in this pull request:

- Add slack notifier to starter.yaml, skip notify for upgrade job, remove the alert manager

**Related issue(s)**

- https://github.com/kyma-project/test-infra/issues/466

### CODE REVIEW

File: prow/cluster/08-prow-addons-ctrl-manager.yaml [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+            secretKeyRef:
+              key: token
+              name: sap-slack-bot-token
+        - name: NOTIFIER_SLACK_REPORTER_CHANNEL
+          value: '#c4core-kyma-ci-force'
+        - name: NOTIFIER_SLACK_REPORTER_ACT_ON_PROW_JOB_TYPE
+          value: periodic,postsubmit
+        - name: NOTIFIER_SLACK_REPORTER_ACT_ON_PROW_JOB_STATE
+          value: failure,error
+        image: eu.gcr.io/kyma-project/test-infra/develop/prow-addons-ctrl-manager:0.0.1
```
Comment (mszostok) [inline prow/cluster/08-prow-addons-ctrl-manager.yaml]: will be change after merging this PR: https://github.com/kyma-project/test-infra/pull/592

File: docs/prow/production-cluster-configuration.md [hunk 2]
```diff
@@ -72,6 +73,7 @@ Use the following tools and configuration:
    - Deploy Prow components with the `a202e595a33ac92ab503f913f2d710efabd3de21`revision.
    - Deploy the Cert Manager.
    - Deploy secure Ingress.
+   - Deploy the [Prow Addons Controller Manager](../../development/prow-addons-ctrl-manager/README.md).
```
Comment (mszostok) [inline docs/prow/production-cluster-configuration.md]: will be visible after merging this PR: https://github.com/kyma-project/test-infra/pull/592

File: docs/prow/production-cluster-configuration.md [hunk 3]
```diff
@@ -16,6 +16,7 @@ Use the following tools and configuration:
 - Two Secrets in the Kubernetes cluster:
   - `hmac-token` which is a Prow HMAC token used to validate GitHub webhooks
   - `oauth-token` which is a GitHub token with read and write access to the `kyma-bot` account
+  - `sap-slack-bot-token` which is a token for publishing messages to the SAP CX workspace. Find more information [here](https://api.slack.com/docs/token-types#bot)
```
Comment (aszecowka) [inline docs/prow/production-cluster-configuration.md]: "Two Secrets" ;)

File: docs/prow/production-cluster-configuration.md [hunk 4]
```diff
@@ -13,9 +13,10 @@ Use the following tools and configuration:
 - [gcloud](https://cloud.google.com/sdk/gcloud/) to communicate with Google Cloud Platform (GCP)
 - The `kyma-bot` GitHub account
 - [Kubernetes cluster](./prow-installation-on-forks.md#provision-a-cluster)
-- Two Secrets in the Kubernetes cluster:
+- Secrets in the Kubernetes cluster:
   - `hmac-token` which is a Prow HMAC token used to validate GitHub webhooks
   - `oauth-token` which is a GitHub token with read and write access to the `kyma-bot` account
+  - `sap-slack-bot-token` which is a token for publishing messages to the SAP CX workspace. Find more information [here](https://api.slack.com/docs/token-types#bot)
```
Comment (kazydek) [inline docs/prow/production-cluster-configuration.md]:
```suggestion
  - `sap-slack-bot-token` which is a token for publishing messages on the SAP CX workspace. Find more information [here](https://api.slack.com/docs/token-types#bot)
```

File: docs/prow/prow-architecture.md [hunk 5]
```diff
@@ -43,6 +43,10 @@ Prow plugins applied for the Kyma project include:
 - **cat** that checks if there is a new GitHub event for a `/meow` comment on a PR. If it finds it, it adds a cat image to the related PR. For that purpose, it uses the GitHub token available as a Kubernetes Secret.
 - **config-updater** that reads the configuration from `config.yaml`, `plugins.yaml`, and the `jobs` folder, and updates it on the production cluster after the merge to the `master` branch. This plugin is only configured for the `test-infra` repository.
 
+## Prow Addons Controller Manager
+
+The Prow Addons Controller Manager contains all custom controller extensions for Prow infrastructure. Currently it contains only the Notifier controller. Notifier watches all ProwJobs and sends notifications to a given Slack channel. Find detailed documentation [here](../../development/prow-addons-ctrl-manager/README.md)
```
Comment (kazydek) [inline docs/prow/prow-architecture.md]:
```suggestion
The Prow Addons Controller Manager contains all custom controller extensions for Prow infrastructure, such as the Notifier controller. Notifier watches all ProwJobs and sends notifications to a given Slack channel. Find detailed documentation [here](../../development/prow-addons-ctrl-manager/README.md)
```

Comment (aszecowka) [inline docs/prow/prow-architecture.md]: maybe make it explicit that it was developed by us, and point to the source code.

**Pull request merged**